src/App.tsx

```diff
@@ -206,27 +206,37 @@ function App() {
             });
 
             const initialSelectedMethods: Record<string, string> = {};
-            Object.keys(result.sessionScopes).forEach((scope) => {
-              initialSelectedMethods[scope] = 'eth_blockNumber';
-
-              const example = metamaskOpenrpcDocument?.methods.find(
-                (method) => (method as MethodObject).name === 'eth_blockNumber',
-              );
-
-              const defaultRequest = {
-                method: 'wallet_invokeMethod',
-                params: {
-                  scope,
-                  request: openRPCExampleToJSON(example as MethodObject),
-                },
-              };
-
-              setInvokeMethodRequests((prev) => ({
-                ...prev,
-                [scope]: JSON.stringify(defaultRequest, null, 2),
-              }));
-            });
+            const initialSelectedAccounts: Record<string, CaipAccountId> = {};
+
+            Object.entries(result.sessionScopes).forEach(
+              ([scope, details]: [string, any]) => {
+                initialSelectedMethods[scope] = 'eth_blockNumber';
+
+                if (details.accounts && details.accounts.length > 0) {
+                  initialSelectedAccounts[scope] = details.accounts[0];
+                }
+
+                const example = metamaskOpenrpcDocument?.methods.find(
+                  (method) =>
+                    (method as MethodObject).name === 'eth_blockNumber',
+                );
+
+                const defaultRequest = {
+                  method: 'wallet_invokeMethod',
+                  params: {
+                    scope,
+                    request: openRPCExampleToJSON(example as MethodObject),
+                  },
+                };
```

Comment:
do we need to store the wrapper? can we add it on later when we make the call in `handleInvokeMethod`? doesn't really matter though

Reply:
yeah... ok if we make this enhancement later?

Reply:
yeah